power_grid_model_c/power_grid_model/include/power_grid_model/main_model_impl.hpp

```diff
@@ -289,50 +280,63 @@ class MainModelImpl<ExtraRetrievableTypes<ExtraRetrievableType...>, ComponentLis
     // helper function to update vectors of components
     template <class CompType, cache_type_c CacheType>
     void update_component(std::vector<typename CompType::UpdateType> const& components,
-                          std::vector<Idx2D> const& sequence_idx) {
+                          std::span<Idx2D const> sequence_idx) {
         if (!components.empty()) {
             update_component<CompType, CacheType>(components.begin(), components.end(), sequence_idx);
         }
     }
     template <class CompType, cache_type_c CacheType>
     void update_component(std::span<typename CompType::UpdateType const> components,
-                          std::vector<Idx2D> const& sequence_idx) {
+                          std::span<Idx2D const> sequence_idx) {
         if (!components.empty()) {
             update_component<CompType, CacheType>(components.begin(), components.end(), sequence_idx);
         }
     }
     template <class CompType, cache_type_c CacheType>
     void update_component(ConstDataset::RangeObject<typename CompType::UpdateType const> components,
-                          std::vector<Idx2D> const& sequence_idx) {
+                          std::span<Idx2D const> sequence_idx) {
         if (!components.empty()) {
             update_component<CompType, CacheType>(components.begin(), components.end(), sequence_idx);
         }
     }
 
+    template <class CompType, cache_type_c CacheType>
+    void update_component(ConstDataset const& update_data, Idx pos, std::span<Idx2D const> sequence_idx) {
+        assert(construction_complete_);
+        assert(update_data.get_description().dataset->name == std::string_view("update"));
+
+        if (update_data.is_columnar(CompType::name)) {
+            this->update_component<CompType, CacheType>(
+                update_data.get_columnar_buffer_span<meta_data::update_getter_s, CompType>(pos), sequence_idx);
+        } else {
+            this->update_component<CompType, CacheType>(
+                update_data.get_buffer_span<meta_data::update_getter_s, CompType>(pos), sequence_idx);
+        }
+    }
+
     // update all components
     template <cache_type_c CacheType>
+    void
+    update_component(ConstDataset const& update_data, Idx pos,
```

Comment:
> Edit: I assume that the last overload is preventing exactly this, right? So such permanent update is still valid.

yes. i deliberately split them up to make it explicit just how many flavors we have. Who knows - this might even fix the macos issue.

> It is confusing, shouldn't we just rename these something like `update_all_components`

i'm ok with that, but i do believe that the overload argument `ConstDataset const& update_data` already signifies this. we can discuss that later or put it off until the refactor after this.

Reply:
Since there are many overloads to the function, it becomes harder to inmediatly determine the difference between them, so making that small change would help achieving that without having to "directly compare" with other overloads, which I found myself doing. But for sure it can be done during refactoring if agreed upon.

Reply:
> It is confusing, shouldn't we just rename these something like update_all_components

I feel the same at many places, we need to read the complete definition to understand what an overload does. Less intuitive.
Maybe whenever we have multiple overloads like this, we should always add a comment / docstring signifying what it does. Just like how is done for this function